validation.go

```diff
@@ -149,7 +149,7 @@ func CheckRecordSig(r *pb.Record, pk ci.PubKey) error {
 	blob := RecordBlobForSig(r)
 	good, err := pk.Verify(blob, r.Signature)
 	if err != nil {
-		return nil
+		return err
```

Comment:
Wow, was this a typo or intentional? We should add comments when doing something as hard to spot (or counter-intuitive) as this.

Reply:
looks like a typo; it must be!

Reply:
I don't see how this is counter intuitive (although the crypto interface is bad, it should just return a single boolean).